src/sdk/namespace_fs.js

```diff
@@ -2069,9 +2077,24 @@ class NamespaceFS {
         }
     }
 
-
+    /**
+     * _delete_single_object does the following before deleting the object
+     * 1. open dir_file and src_file fd
+     * 2. _unlink_file_ignore_enoent - which means it stats the to be deleted file, validate filter if exists, unlink safely
+     * 3. deleted parent directories if they are empty 
+     * 4. clears directory object xattr if relevant
+     * 5. closes file and dir_file
+     */
     async _delete_single_object(fs_context, file_path, params) {
-        await native_fs_utils.unlink_ignore_enoent(fs_context, file_path);
```

Comment:
I think we should have a separation between the lifecycle path that requires safe_unlink and general path that doesn't. the extra file openings and validations are not required for cases other than lifecycle